pkg/apis/autoscaling/validation/validation.go

```diff
@@ -26,6 +27,12 @@ import (
 	"k8s.io/kubernetes/pkg/apis/autoscaling"
 )
 
+var (
+	invalidTargets = []string{
```

Comment:
How about having a `map[string]struct{}` so that we can avoid the `for` and simply lookup ?